### PR TITLE
Expose the dataTracker composable to enable its use in third-party nodes

### PR DESCRIPTION
**Pull request by cstns**
## Description

Expose the dataTracker composable to enable its use in third-party nodes

## Related Issue(s)

closes https://github.com/FlowFuse/node-red-dashboard/issues/1076

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: ui/src/main.mjs [hunk 1]
```diff
@@ -41,6 +41,7 @@ const theme = {
         error: '#ff5252'
     }
 }
+import { useDataTracker } from './widgets/data-tracker.mjs' // eslint-disable-line import/order
```
Comment (joepavitt) [inline ui/src/main.mjs:44]: linting error FYI

**Comment by cstns** [comment]
Should be merged before https://github.com/FlowFuse/node-red-dashboard/pull/1097

**Comment by joepavitt** [comment]
FYI @cstns you have _a lot_ of failing E2E's here

**Comment by cstns** [comment]
When running locally the only suite that fails is the widgets/form.spec.js, but I'm getting the same result on the main branch. 

I'm still looking into causes that might cause this behavior. Seems to be a difference in behavior in CI an local when running e2e tests. Could be related to https://github.com/FlowFuse/node-red-dashboard/issues/882 but can't say for sure yet.

**Comment by cstns** [comment]
changed source branch to https://github.com/FlowFuse/node-red-dashboard/pull/1110 to see how ci e2e tests preform

File: cypress.config.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ module.exports = defineConfig({
         specPattern: 'cypress/tests/**/*.spec.{js,jsx,ts,tsx}',
         setupNodeEvents (on, config) {
             // implement node event listeners here
-        }
+        },
+        defaultCommandTimeout: 10000,
+        retries: { runMode: 3, openMode: 1 }
```
Comment (joepavitt) [inline cypress.config.js:11]:
```suggestion
        }
```

Comment (joepavitt) [reply cypress.config.js:11]: we shouldn't be relying on this to have passing tests, it also slows the suite down considerably

**Comment by joepavitt** [comment]
Screenshot 2024-07-26 at 17 21 05

Errors appearing for `ui-template` node

**Review by joepavitt — APPROVED** [review]

**Pull request merged**